src/axolotl/prompt_strategies/chat_template.py

```diff
@@ -424,61 +441,82 @@ def find_turn(self, turns: list[dict], turn_idx: int):
 
     def get_conversation_thread(self, prompt):
         turns = []
-        optional_keys = [
-            "tool_calls",  # tool that 'assistant' calls
-            "name",  # name of tool given by 'tool'
-            "tool_call_id",  # mistral/mixtral requires this
-        ]
         for message in prompt[self.messages]:
+            transformed_message = self.transform_message(message)
+
             turn = {
-                "role": self.prompter.roles[message[self.prompter.message_field_role]],
+                **transformed_message,
```

Comment:
Would we need to worry about any nested copying here? Should we set `turn = transformed_message` then set the other two fields after?